state/home/actions.ts

```diff
@@ -0,0 +1,17 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { providers } from 'ethers';
+
+import { FuturesMarketSerialized } from 'sdk/types/futures';
+import { ThunkConfig } from 'state/types';
+import { serializeMarkets } from 'utils/futures';
+
+export const fetchOptimismMarkets = createAsyncThunk<
+	{ markets: FuturesMarketSerialized[] },
+	providers.Provider,
+	ThunkConfig
+>('home/fetchOptimismMarkets', async (mainnetL2Provider, { extra: { sdk } }) => {
+	// For the home page we always fetch OP mainnet markets
+	const markets = await sdk.futures.getMarkets({ provider: mainnetL2Provider, networkId: 10 });
```

Comment:
Instead of passing a provider to this method, what do you think about creating an L2 provider instance on the SDK itself, since we might probably need it in other cases in the future.

Reply:
Yea might make sense, was thinking it could be a bit inefficient to keep two instances of the sdk around with one of them only being used on rare occasions. Seems this is the only place it's used right now so could revisit when there are more use cases?

Reply:
Yeah. Fair point. It might be a premature optimization. Maybe we can make a note in the SDK readme for the future though.